### PR TITLE
Add class diagram for `DarcLib` git clients

### PR DESCRIPTION
**Pull request by premun**
https://github.com/dotnet/arcade-services/issues/3014

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements

### Release Note Description
Not release notes worthy

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/DarcLib/Architecture.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+This document describes some of the architecture of the code in `DarcLib`, especially classes for managing git.
+
+### Git clients
+
+The `DarcLib` has several git-related classes which manage either a local git directory or a remote GitHub/AzureDevOps repository.
```
Comment (tkapin) [inline src/Microsoft.DotNet.Darc/DarcLib/Architecture.md:5]: Can you add one sentence explaining why we need several clients? This is the first question that any reader would have here. Cheers!

**Review by tkapin — DISMISSED** [review]

File: src/Microsoft.DotNet.Darc/DarcLib/Architecture.md (hunk 1 above)
Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/Architecture.md:5]: Just pushed a commit

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/Architecture.md:5]: Ah, will add one more

Comment (premun) [reply src/Microsoft.DotNet.Darc/DarcLib/Architecture.md]: @tkapin can you re-approve please?

**Comment by premun** [comment]
@tkapin gentle ping, can you re-approve please?

**Comment by premun** [comment]
FYI I also added exclusion of build runs for documentation changes

**Review by tkapin — APPROVED** [review]

**Pull request merged**